crates/control/src/motion/joint_command_collector.rs

```diff
@@ -0,0 +1,116 @@
+        let head_joints_command = context.head_joints_command;
+        let motion_selection = context.motion_selection;
+        let arms_up_squat = context.arms_up_squat_joints_command;
+        let jump_left = context.jump_left_joints_command;
+        let jump_right = context.jump_right_joints_command;
+        let sit_down = context.sit_down_joints_command;
+        let stand_up_back_positions = context.stand_up_back_positions;
+        let stand_up_front_positions = context.stand_up_front_positions;
+        let walk = context.walk_joints_command;
+
+        let (positions, stiffnesses) = match motion_selection.current_motion {
+            MotionType::ArmsUpSquat => (arms_up_squat.positions, arms_up_squat.stiffnesses),
+            MotionType::Dispatching => (
+                dispatching_command.positions,
+                dispatching_command.stiffnesses,
+            ),
+            MotionType::FallProtection => (fall_protection_positions, fall_protection_stiffnesses),
+            MotionType::JumpLeft => (jump_left.positions, jump_left.stiffnesses),
+            MotionType::JumpRight => (jump_right.positions, jump_right.stiffnesses),
+            MotionType::Penalized => (*context.penalized_pose, Joints::fill(0.8)),
+            MotionType::SitDown => (sit_down.positions, sit_down.stiffnesses),
+            MotionType::Stand => (
+                Joints::from_head_and_body(head_joints_command.positions, walk.positions),
+                Joints::from_head_and_body(head_joints_command.stiffnesses, walk.stiffnesses),
+            ),
+            MotionType::StandUpBack => (*stand_up_back_positions, Joints::fill(1.0)),
+            MotionType::StandUpFront => (*stand_up_front_positions, Joints::fill(1.0)),
+            MotionType::Unstiff => (current_positions, Joints::fill(0.0)),
+            MotionType::Walk => (
+                Joints::from_head_and_body(head_joints_command.positions, walk.positions),
+                Joints::from_head_and_body(head_joints_command.stiffnesses, walk.stiffnesses),
+            ),
+            MotionType::EnergySavingStand => (
+                Joints::from_head_and_body(
+                    head_joints_command.positions,
+                    context.energy_saving_stand_command.positions,
+                ),
+                Joints::from_head_and_body(
+                    head_joints_command.stiffnesses,
+                    context.energy_saving_stand_command.stiffnesses,
+                ),
+            ),
+        };
+
+        // The actuators uses the raw sensor data (not corrected like current_positions) in their feedback loops,
+        // thus the compensation is required to make them reach the actual desired position.
+        let compensated_positions = positions + *context.joint_calibration_offsets;
+        let collected_commands = CollectedCommands {
+            positions,
+            compensated_positions,
+            stiffnesses,
+            leds: *context.leds,
```

Comment:
Since the idea of ```CollectedCommands``` is, that it includes everything that is written into the hardware interface. The name is maybe confusing

Reply:
This concept is not very strictly implemented, because you are modifying things in the `CollectedCommands`, which means not all things are written into the hardware interface. It would be better if you would not duplicate e.g. LEDs because they are already contained in the database. The last node could take them from the `CollectedCommands` but it is confusing because it could also use `Input<Leds>`. 🤔

Reply:
Okay agree, removed LEDs from `CollectedCommands`